CHANGELOG.md

```diff
@@ -6,6 +6,10 @@
 
 - XCFrameworks are now signed (#5271)
 
+### Improvements
+
+- Remove unnecessary lock in SentryLog.configure (#5297)
```

Comment:
`m`: The changelog should mention the SDK user benefit, so maybe add something like:

```suggestion
- Reduce performance impact by removing unnecessary lock in SentryLog.configure (#5297)
```